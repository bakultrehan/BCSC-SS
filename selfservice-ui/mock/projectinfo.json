{
  "project-info": [
    {
      "id": 1,
      "organizationName": "organic",
      "projectName": "project",
      "myRole": "1",
      "developerDetails": {
        "email": "test@test.com",
        "phone": "0987654321",
        "firstName": "name",
        "lastName": "l name",
        "id": 1
      },
      "managerDetails": {
        "email": "mail@test.com",
        "phone": "1234567890",
        "firstName": "f manager",
        "lastName": "l manager",
        "id": 2
      },
      "ctoDetails": {
        "email": "cto@email.com",
        "phone": "1234567890",
        "firstName": "f cto",
        "lastName": "l cto",
        "id": 3
      }
    },
    {
      "organizationName": "aa",
      "projectName": "aa",
      "description": "testaa",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "ml",
        "email": "me",
        "phone": "mp"
      },
      "ctoDetails": {
        "firstName": "cf",
        "lastName": "cl",
        "email": "ce",
        "phone": "cp"
      },
      "id": 11
    },
    {
      "organizationName": "ss1",
      "projectName": "dd",
      "description": "test",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "m",
        "email": "m",
        "phone": "m"
      },
      "ctoDetails": {
        "firstName": "c",
        "lastName": "c",
        "email": "c",
        "phone": "c"
      },
      "id": 12
    }
  ]
}