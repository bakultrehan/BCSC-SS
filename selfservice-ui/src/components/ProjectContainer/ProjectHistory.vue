/** * ProjectHistory of app */

<template>
  <v-card class="mx-auto">
    <v-container>
      <v-row class="ma-2" v-if="isLoading">
        <Loading />
      </v-row>
      <v-row class="ma-2" v-else>
        <v-col cols="12">
          <v-col cols="12" md="12" v-if="projectHistoryData.length > 0">
            <h2 class="bc-container-title-h1 px-2 mb-5">
              {{ $t('projectHistory.tabSubTitle') }}
            </h2>

            <v-simple-table class="text-left">
              <template v-slot:default>
                <thead class="table-head">
                  <tr>
                    <th :scope="$t('projectHistory.tblTitleProjectStatus')">
                      {{ $t('projectHistory.tblTitleProjectStatus') }}
                    </th>
                    <th :scope="$t('projectHistory.tblTitleDateAssigned')">
                      {{ $t('projectHistory.tblTitleDateAssigned') }}
                    </th>
                    <th :scope="$t('projectHistory.tblActionTakenBy')">
                      {{ $t('projectHistory.tblActionTakenBy') }}
                    </th>
                    <th :scope="$t('projectHistory.tblActionRole')">
                      {{ $t('projectHistory.tblActionRole') }}
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    v-for="history in projectHistoryData"
                    :key="history.id"
                    @click="redirectToProject(project)"
                    style="cursor: pointer"
                  >
                    <td>
                      {{
                        $t(
                          `projectStatus.status${
                            projectStatusList[history.status]
                          }`
                        )
                      }}
                    </td>
                    <td>{{ history.created }}</td>
                    <td>{{ history.name }}</td>
                    <td>
                      {{
                        $t(
                          history.role &&
                            `projectRoles.role${projectRolesList[history.role]}`
                        )
                      }}
                    </td>
                  </tr>
                </tbody>
              </template>
            </v-simple-table>
          </v-col>
          <v-col cols="12" md="12" v-else>
            <v-card flat class="text-left">
              <v-card-text>{{ $t('projectHistory.noData') }}</v-card-text>
            </v-card>
          </v-col>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { projectStatus, projectRoles } from '@/constants/enums';
import VirtualCardCount from '@/components/Dashboard/VirtualCardCount.vue';
import Loading from '@/Atomic/Loading/Loading.vue';

import Button from '@/Atomic/Button/Button.vue';

const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: {
    Button,
    VirtualCardCount,
    Loading,
  },
})
export default class ProjectHistory extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @ProjectInfoModule.Getter('getProjectHistory')
  public getProjectHistory!: any;
  @ProjectInfoModule.Action('loadProjectHistory')
  public loadProjectHistory!: any;

  private projectStatusList: any = projectStatus;
  private projectRolesList: any = projectRoles;
  private isLoading: boolean = true;
  private projectHistoryData: any = [];

  @Watch('getProjectHistory')
  private ongetProjectHistoryChanged(val: any) {
    this.isLoading = false;
    this.projectHistoryData = val;
  }

  private mounted() {
    this.loadProjectHistory(this.id);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.table-head {
  background-color: $BCgovBlue5 !important;
  color: $BCgovWhite !important;
  & tr {
    & th {
      color: $BCgovWhite !important;
      font-size: 14px;
    }
  }
}
</style>
