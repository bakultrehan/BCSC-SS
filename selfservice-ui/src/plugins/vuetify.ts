import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
