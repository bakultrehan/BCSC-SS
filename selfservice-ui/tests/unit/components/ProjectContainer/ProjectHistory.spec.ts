import { shallowMount, createLocalVue } from '@vue/test-utils';
import ProjectHistory from '@/components/ProjectContainer/ProjectHistory.vue';
import vuetify from 'vuetify';
import Vuex from 'vuex';
import { singleProject } from '../../mocks/projectinfo';

describe('ProjectHistory.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);
    const store = new Vuex.Store({
      modules: {
        ProjectInfoModule: {
          namespaced: true,
          state: {},
          getters: {
            getProjectHistory: jest.fn(() => {
              return { singleProject };
            }),
          },
          actions: {
            loadProjectHistory: jest.fn(),
          },
        },
      },
    });

    wrapper = shallowMount(ProjectHistory, {
      localVue,
      store,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
