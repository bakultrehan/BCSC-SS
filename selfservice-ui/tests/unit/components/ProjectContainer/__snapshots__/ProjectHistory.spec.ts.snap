// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectHistory.vue renders props when passed 1`] = `
<v-card-stub
  class="mx-auto"
  loaderheight="4"
  tag="div"
>
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      class="ma-2"
      tag="div"
    >
      <loading-stub />
    </v-row-stub>
  </v-container-stub>
</v-card-stub>
`;
